Formula/a/aws-sam-cli.rb

```diff
@@ -17,48 +17,113 @@ class AwsSamCli < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "944427377be6c2a0ead2f53438157210c3afd48ac2656e4199ee8af28c38793c"
   end
 
-  depends_on "rust" => :build # for rpds-py
-  depends_on "cffi"
-  depends_on "cfn-lint"
-  depends_on "cookiecutter"
-  depends_on "pycparser"
-  depends_on "pygments"
-  depends_on "python-cryptography"
```

Comment:
I thought we wanted to keep at least `python-cryptography` since it needs rust to build? https://github.com/Homebrew/homebrew-core/issues/157500#issuecomment-1928947480

Also, I would rev bump here since this fixes a bug

Reply:
At AGM it was discussed and decided that we are going to remove all of the formulae that we can otherwise have as resources from PyPI. It's been causing issues with formulae breaking, incompatibility with brew-pip-audit, etc. I've added you to a channel in Slack about this project.

Reply:
I could go either way here. It'd be good to reuse `python-cryptography` in that way but then this formula has a `depends_on "rust" => :build` here anyway so there's less difference 🤷🏻

Reply:
`python-cryptography` is getting removed as well. Think it's better to have a clean break here instead of ambiguity.